tests/testing_utils_test.go

```diff
@@ -17,8 +17,10 @@ func runWithTimeout(
 		timeoutEvent := time.After(timeout)
 		done := make(chan bool)
 		go func() {
+			defer func() {
+				done <- true
+			}()
```

Comment:
I thought you had to insert a `recover` statement here, but upon reading the [defer/panic/recover](https://go.dev/blog/defer-panic-and-recover) post, I see that I was wrong. 

```golang
			defer func() {
			        if r := recover(); r != nil {
			            t.Error("run failed.")
                                    done <- true
                                }				
			}()
```

I believe your code is correct as is.

Reply:
I was empirically running into hangs that were fixed by this. Definitely learned something about Go also - defer is taking panics into account as well as early (error) returns.